tests/test_qsh.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+from jira.client import QshGenerator
+
+
+class MockRequest(object):
+    def __init__(self, method, url):
+        self.method = method
+        self.url = url
+
+
+def test_qsh():
+    gen = QshGenerator("http://example.com")
+
+    for method, url, expected in [
```

Comment:
Awesome test cases, let's arrange them into a pytest friendly way !

Specifically with a [`@pytest.mark.parametrize`](https://docs.pytest.org/en/6.2.x/parametrize.html)

An example can be found here:
https://github.com/pycontribs/jira/pull/1109/files#diff-3ce874fa74c2993e05bfbc5fbc2df93e0d043ab22ae5dd0c1f7c4f0ce56e4216R13

So you would just be pulling out the for loop into the function argument.

```python
@pytest.mark.parametrize(
    "method,url,expected",
     [("GET", "http://example.com", "GET&&"), ...],
    ids=["empty parameter", ...]
)
def test_qsh(method, url, expected): ...
```

Reply:
Sure! I've made the suggested change.